signify-ts-test/test/multi-user-api-permissions.test.ts

```diff
@@ -0,0 +1,396 @@
+        user.uploadDig = signedZipDig;
+        break;
+      }
+    }
+  }
+  // check upload by aid
+  let sresp = await getReportStatusByAid(
+    env.roleName,
+    user1.ecrAid.prefix,
+    user1.roleClient,
+  );
+  assert.equal(sresp.status, 202);
+  let sbody = await sresp.json();
+  assert.equal(sbody.length, 1);
+
+  // check upload by aid from different lei
+  sresp = await getReportStatusByAid(
+    env.roleName,
+    user3.ecrAid.prefix,
+    user1.roleClient,
+  );
+  assert.equal(sresp.status, 401);
+
+  // check upload by dig from different lei
+  sresp = await getReportStatusByDig(
+    env.roleName,
+    user3.ecrAid.prefix,
+    user3.uploadDig,
+    user1.roleClient,
+  );
+  assert.equal(sresp.status, 401);
+
+  // check upload by dig from the same lei
+  sresp = await getReportStatusByDig(
+    env.roleName,
+    user1.ecrAid.prefix,
+    user2.uploadDig,
+    user1.roleClient,
+  );
+  assert.equal(sresp.status, 200);
+
+  // check LEI upload statuses by aid
+  sresp = await getLeiReportStatusesByAid(
+    env.roleName,
+    user1.ecrAid.prefix,
+    user1.roleClient,
+  );
+  assert.equal(sresp.status, 202);
+  sbody = await sresp.json();
+  assert.equal(sbody.length, 2);
+}, 200000);
+
+export async function getGrantedCredential(
+  client: SignifyClient,
+  credId: string,
+): Promise<any> {
+  const credentialList = await client.credentials().list({
+    filter: { "-d": credId },
+  });
+  let credential: any;
+  if (credentialList.length > 0) {
+    assert.equal(credentialList.length, 1);
+    credential = credentialList[0];
+  }
+  return credential;
+}
+
+async function getReportStatusByAid(
+  aidName: string,
+  aidPrefix: string,
+  client: SignifyClient,
+): Promise<Response> {
+  const heads = new Headers();
+  const sreq = { headers: heads, method: "GET", body: null };
+  const surl = `${env.apiBaseUrl}/status/${aidPrefix}`;
+  let shreq = await client.createSignedRequest(aidName, surl, sreq);
+  const sresp = await fetch(surl, shreq);
+  return sresp;
+}
+
+async function getLeiReportStatusesByAid(
+  aidName: string,
+  aidPrefix: string,
+  client: SignifyClient,
+): Promise<Response> {
+  const heads = new Headers();
+  const sreq = { headers: heads, method: "GET", body: null };
+  const surl = `${env.apiBaseUrl}/report/status/lei/${aidPrefix}`;
+  let shreq = await client.createSignedRequest(aidName, surl, sreq);
+  const sresp = await fetch(surl, shreq);
+  return sresp;
+}
+
+async function getReportStatusByDig(
+  aidName: string,
+  aidPrefix: string,
+  dig: string,
+  client: SignifyClient,
+): Promise<Response> {
+  const heads = new Headers();
+  const sreq = { headers: heads, method: "GET", body: null };
+  const surl = `${env.apiBaseUrl}/upload/${aidPrefix}/${dig}`;
+  let shreq = await client.createSignedRequest(aidName, surl, sreq);
+  const sresp = await fetch(surl, shreq);
+  return sresp;
+}
+
+async function dropReportStatusByAid(aidPrefix: string): Promise<Response> {
```

Comment:
We need to consolidate these functions since they are being repeated. i started that with my PR but we need to do it for calls like these as well.